### PR TITLE
Add optionalkey parsing

### PR DESCRIPTION
**Pull request by nathunsmitty**
Parses a new AVDL decorator, `@optional(true)`, in struct fields. With this field, properties are marked with `omitempty`. Also adds a test for this!

### CODE REVIEW

File: test/files/sample.go [hunk 1]
```diff
@@ -4,56 +4,61 @@
 package sample1
 
 import (
```
Comment (nathunsmitty) [inline test/files/sample.go]: I think `gofmt` took a hold of this file, which is why the diff is so big. Only addition should be the `Woop` field on `R`.

**Comment by nathunsmitty** [comment]
@maxtaco bump on this 🙂

**Comment by nathunsmitty** [comment]
Just because the version is bumped to v1.4.1

**Comment by maxtaco** [comment]
ok. better to let me bump the version as part of the release

**Comment by nathunsmitty** [comment]
If you're cutting a new release, could we also squeeze #29 in there too?

**Pull request merged**